lib/graphql/query/serial_execution/field_resolution.rb

```diff
@@ -16,14 +16,14 @@ def initialize(ast_node, parent_type, target, query, execution_strategy)
 
         def result
           result_name = ast_node.alias || ast_node.name
-          result_value = begin
-            get_finished_value(get_raw_value)
+          raw_value = begin
+            get_raw_value
           rescue GraphQL::ExecutionError => err
             err.ast_node = ast_node
             query.context.errors << err
-            nil
+            err
           end
-          { result_name => result_value  }
+          { result_name => get_finished_value(raw_value) }
```

Comment:
Yes, I guess `GraphQL::ExecutionError` wouldn't be appropriate for errors in type resolution or result coercion, since they would be more like the `nil` on a non-null field error that this PR addresses.